system/services/siteTree/SiteTreeService.cfc

```diff
@@ -189,12 +189,15 @@ component {
 		, array   ids          = []
 		, array   extraFilters = []
 	) {
-		var filter = { "page.trashed" = false };
-		var extra  = Duplicate( arguments.extraFilters );
-		var params = {};
+		var filter  = { "page.trashed" = false };
+		var extra   = Duplicate( arguments.extraFilters );
+		var params  = {};
+		var orderBy = "page._hierarchy_sort_order";
 
 		if ( arguments.ids.len() ) {
 			extra.append( { filter={ "page.id"=arguments.ids } } );
+			params.ids = { list=true, type="varchar", value=arguments.ids }
+			orderBy = "FIELD( page.id, :ids )"
```

Comment:
Please note that `FIELD()` is MySQL-specific and not supported by MSSQL, so this code will fail if run in MSSQL...